### ISSUE TITLE
RDF UX incubation / RDF UI replacement 

### ISSUE THREAD

**Issue by DiegoPino** [Islandora-CLAW/CLAW#647]
Hi,

One of the current needs CLAW has is the ability to manage via UI (right now most if that happens via config entities) other Ontologies, rdf:types and predicates than the ones RDF UI via schema.org provides. 

RDF UI for Drupal 8 was a one person/summer project and even if we could transfer that responsibility to us and maintain it, the code needs some heavy rewriting. So I will start a new project named RDF UX, that will go into islandora-labs, incubate and grow there to become, eventually, part of CLAW

To keep the complexity of this in scope of MVP the base module will handle most of the needs identified by @kimpham54 and the Islandora metadata IG 

which means:

- https://github.com/Islandora-CLAW/CLAW/issues/489
- https://github.com/Islandora-CLAW/CLAW/issues/486
- https://github.com/Islandora-CLAW/CLAW/issues/487
- https://github.com/Islandora-CLAW/CLAW/issues/488
- https://github.com/Islandora-CLAW/CLAW/issues/489
- 

I also identified a few different things this module will need, from Web/Javascript integrations test to integrating into config entity export functionality. I will also add an extendable framework so others can build semantic aware modules that make more powerful use of LoD in the future. Since it will go into labs, issue tracking will happen there probably.

The main part of the development will start 2 weeks after iCamp Europe and anyone is welcome to contribute of course.

**Comment by dannylamb** [comment]
That's fantastic @DiegoPino.  I'll set you up with a repository in Labs and give you the perms you need. I can also 'transfer' those use cases (a.k.a. copy/paste those issues) for you too.

**Comment by dannylamb** [comment]
@DiegoPino https://github.com/Islandora-Labs/rdfux

**Comment by DiegoPino** [comment]
@dannylamb thanks. No worries about the issue, I can "move" them later when needed. Probably more a modal copy, so the original ones stay in claw? we can decide that later of course

**Comment by dannylamb** [comment]
@DiegoPino Oh, sorry.  I already started turning the crank on those issues.  I'm linking back to the originals on them.  That ok with you?

Otherwise I'll gladly re-open the one issue I already closed.  No worries.

**Comment by DiegoPino** [comment]
@dannylamb np! Just wanted to avoid you spending time on that. thanks!

**Comment by dannylamb** [comment]
@DiegoPino Not a problem.

When you're ready to start work after ICampEU, we can coordinate some messaging on the list to drum up engagement for you.

**Comment by whikloj** [comment]
@DiegoPino I was looking at the RDF-UI module and it appears that they load the Schema.org graph from [the website](http://schema.org/docs/schema_org_rdfa.html) (and cache it locally). That seems like it might be good for some of the more common ontologies. It appears that the [dcterms ontology webpage](http://dublincore.org/documents/2012/06/14/dcmi-terms/?v=terms) is marked up in much the same way as Schema.org so we could conceivably load it using EasyRdf as well.

**Comment by DiegoPino** [comment]
@whikloj yes, they do. I'm inclined to do the same, or at least, the same approach on RDF-UX. Thanks!

**Comment by whikloj** [comment]
@DiegoPino just wondering where you are with this, I know you have lots on your plate but I started (like just started today) to build a way to load ontologies. But if you have something started I can work off yours?

**Comment by DiegoPino** [comment]
@whikloj probably yours will be closer to CLAW needs and current Road Map, so feel free to take over. Eventually, probably after LDCX/same time, I get 7x. RC1 is out, I will be able to share my development with you all. Thanks

**Comment by dannylamb** [comment]
Dusting this off the shelf and making it a Roadmap issue.  Initially we thought development would be done in a separate module, which we [created in labs](https://github.com/Islandora-Labs/rdfux) to hopefully foster development.  The issues were moved to  https://github.com/Islandora-Labs/rdfux .

Of the original features given consideration by @DiegoPino and the MIG, many are handled by the way we use taxonomy terms and external uris.  What we are lacking now is really the UI aspect of it.  I'll list out all of the existing issues and their level of completion.

- ~https://github.com/Islandora-Labs/rdfux/issues/1~ 
  - The way we are now modeling authorities with `controlled_access_terms` satisfies this issue/use case.
- https://github.com/Islandora-Labs/rdfux/issues/2
  - I'm not sure if I'm interpreting this correctly, but by the looks of it, this issue is really saying "Make the UI itself"
- https://github.com/Islandora-Labs/rdfux/issues/3
  - This will be resolved with https://github.com/Islandora-CLAW/CLAW/issues/889
- https://github.com/Islandora-Labs/rdfux/issues/4
  - I'd put this at low priority, since having a functional UI should take precedence.
- ~https://github.com/Islandora-Labs/rdfux/issues/5~
  - We are currently doing this using Drupal's migrate functionality in core Islandora.  So we have vetted examples of doing this, we just need documentation, really.

So two down, one on the way, and the next issue to get worked on will be the UI itself.

**Comment by rosiel** [comment]
| Title (Goal)  | RDF Mapping UI                            |
--------------- |------------------------------------                           |
| Primary Actor | Collection Manager, Site Administrator  |
| Scope         | RDF Config       |
| Story         | "s a site administrator I wish to have a more intuitive way of mapping RDF via a drop-down matching interface.

Source: Islandora Features Survey

**Comment by forresthopkinsa** [comment]
This remains an important UX item in my opinion
